metaflow/runtime.py

```diff
@@ -134,6 +134,10 @@ def __init__(
         # finished.
         self._control_num_splits = {}  # control_task -> num_splits mapping
 
+        # to allow the current singleton access within Tasks (which are in subprocesses)
+        os.environ["CURRENT_MAX_WORKERS"] = str(max_workers)
+        os.environ["CURRENT_MAX_NUM_SPLITS"] = str(max_num_splits)
```

Comment:
Would this work with AWS Batch or Kubernetes?

Reply:
@sam-watts any updates? we would like to ship this PR soon.

Reply:
@savingoyal hoping to get back to this within the next week